File: 1658-min-operations/solution.py
from typing import List
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        total_sum = sum(num for num in nums)
        target = total_sum - x
        left, cur_sum = 0, 0
        for right in range(0, len(nums)):
            cur_sum += nums[right]
            if cur_sum == target:
                return len(nums) - right
            if cur_sum > target:
                cur_sum -= nums[left]
        return -1
if __name__ == "__main__":
    nums = [1, 1, 4, 2, 3]
    x = 5
    print("Ans = ", Solution().minOperations(nums, x))

File: 1658-min-operations/solution.go
package main

import (
	"fmt"
	gft "gopractice/gofunctool"
)

func minOperations(nums []int, x int) int {
	sum := gft.ArrSumInt(nums)
	if x > sum {
		return -1
	}
	target := sum - x

	l := 0
	curVal, maxVal := 0, -1
	for r := 0; r < len(nums); r++ {
		curVal += nums[r]
		fmt.Println("c = ", curVal)
		for curVal > target {
			curVal -= nums[l]
			l++
		}
		if curVal == target {
			maxVal = max(maxVal, r-l+1)
		}
	}
	if maxVal == -1 {
		return -1
	} else {
		return len(nums) - maxVal
	}
}
func main() {
	nums := []int{1, 1, 1, 1}
	x := 4
	fmt.Println("Ans = ", minOperations(nums, x))
}

/*
執行代碼
go run ./1658-min-operations

寫第二次了, 卡在上面這個test case寫不出來
一開始寫了第16行初始化maxVal = 0, 第28行if maxVal == 0, 導致上面這test case一直跑不過
修正成現在這樣才對
*/
